### DISCUSSION TITLE
How to change the variation rate in sensitivity analysis?

### DISCUSSION THREAD

**Discussion by aryaman3900** [biomass-dev/biomass#80] [Q&A] [answered]
Hi @himoto,

I am trying to change the variation rate in sensitivity analysis, which is originally set to 1% to 0.1%. Could you please tell me from where it has to be done? I tried to explore and found that the rate is mentioned in reaction.py as 1.01. I changed it to 1.001 but the results of sensitivity analysis remained the same. Could you please tell me if having the same results is possible or I am doing it the wrong way?

**Comment by himoto** [comment, marked as answer]
Hi @aryaman3900,

The sensitivity of each parameter is defined by:
_C_i = d ln(M) / d ln(p_i)_
where p_i is the i-th parameter, p is parameter vector p = (p_1, p_2, …) and M is a signaling metric, e.g., time-integrated response.
To calculate this, we use finite difference approximations with 1% (=1.01) changes in the parameter values.

> I tried to explore and found that the rate is mentioned in reaction.py as 1.01. I changed it to 1.001 but the results of sensitivity analysis remained the same.

This sounds reasonable to me because 1% change is already enough to approximate the small displacement. I guess even if you use 5% change (=1.05), the results will be quite similar.
Currently, there is no parameter to modify the variation rate, but a future version might have it.

**Reply by aryaman3900** [reply]
Sure @himoto, Thank You very much for confirming this.
